### ISSUE TITLE
Target size and tensor size mismatch in the inference

### ISSUE THREAD

**Issue by arupmondal835** [kyonofx/scdp#2]
Hello, 

I am trying to run the inference step with one of the trained models and I am getting the following mismatch error. 

-------------------------------------------
(/home/amondal2/.conda-envs/scdp) [amondal2@kd1 scdp] GPU $ python scdp/scripts/test.py --data_path ./lmdb  --split_file ./split.json --tag test --ckpt_path ./share_models/qm9_bond_K8L6_beta_1.3_vo
Seed set to 42
/home/amondal2/.local/lib/python3.10/site-packages/torch/jit/_check.py:181: UserWarning: The TorchScript type system doesn't support instance-level annotations on empty non-base types in `__init__`. Instead, either 1) use a type annotation in the class body, or 2) wrap the type in `torch.jit.Attribute`.
  warnings.warn("The TorchScript type system doesn't support "
  0%|                                                                                                           | 0/3 [00:00<?, ?it/s]
  0%|                                                                                                           | 0/3 [00:12<?, ?it/s]
Traceback (most recent call last):
  File "/kfs3/scratch/amondal2/scdp/scdp/scripts/test.py", line 224, in <module>
    main(args.ckpt_path, args.data_path, args.split_file, args.tag, 
  File "/kfs3/scratch/amondal2/scdp/scdp/scripts/test.py", line 149, in mainpt_path ./share_models/qm9_bond
    coeffs, expo_scaling = model.predict_coeffs(batch)
  File "/kfs3/scratch/amondal2/scdp/scdp/model/module.py", line 98, in predict_coeffs
    batch_n_orbs = scatter(n_orbs, batch.batch, len(batch))
  File "/kfs3/scratch/amondal2/scdp/scdp/common/utils.py", line 19, in scatter
    index = index.reshape(-1, *([1]*(len(src.shape)-1))).expand_as(src)
RuntimeError: The expanded size of the tensor (57) must match the existing size (91) at non-singleton dimension 0.  Target sizes: [57].  Tensor sizes: [91]
---------------------------------------------

Do you know where the error is coming from and how I can approach to fix it? Also, I am not sure why the dataset split file is even needed for the inference stage and if it is needed then how I should format it. Here, I have 3 lmdb files in the lmdb directory. So I made up the file as "{"test":[0, 1, 2]}". Is that the right format?

Thank you!
Arup

**Comment by kyonofx** [comment]
Hi,

Thanks for your interest. If you followed every step in the readme you should only have one directory containing all lmdbs, then the split file specify which data objects to be used for testing.

Looking at your error message my guess is you might have preprocessed without vnodes. the pretrained model you are using seem to be the model with bond-vnode -- have you ran the preprocessing script with `--vnode_method bond` as described in the README?

Also, the pretrained model will likely only work for the QM9 dataset it was trained on.

**Comment by arupmondal835** [comment]
Thanks for your reply. 

Yes, I did do the preprocessing step to convert to lmdb with the following command as mentioned in the readme and it ran successfully. 


`python scdp/scripts/preprocess.py --disable_pbc --data_path dataset/ --out_path lmdb_bondvnode --tar --device cpu --atom_cutoff 6 --vnode_method bond`


This command created two files: `data.0000.lmdb` and  `data.0000.lmdb-lock`. Then I ran the following command and got the previous mismatch error. 


`python scdp/scripts/test.py --data_path lmdb_bondvnode/  --split_file split.json --tag test --ckpt_path share_models/qm9_bond_K8L6_beta_1.3_vo/`


I think I am following every step as mentioned in the readme except the split.json file which is not properly mentioned how to make. So looking at the given split file for the training stage, I just made this up as follows:

```
(/home/amondal2/.conda-envs/scdp) [amondal2@kd1 scdp] GPU $ cat split.json 
{"test":[0, 1, 2]}
```

I used 0,1,2 as I have 3 systems in the initial tarball. Does it look correct?
